mgmt_upgrade_test.py

```diff
@@ -0,0 +1,109 @@
+        config_file = '/etc/scylla-manager-agent/scylla-manager-agent.yaml'
+        region_name = self.params.get('region_name').split()[0]
+        for node in self.db_cluster.nodes:
+            update_config_file(node=node, region=region_name, config_file=config_file)
+        sleep(60)
+
+
+def wait_until_task_finishes_return_details(task, wait=True, timeout=1000, step=10):
+    if wait:
+        task.wait_for_status([TaskStatus.DONE], timeout=timeout, step=step)
+    task_history = task.history
+    latest_run_id = task.latest_run_id
+    task_details = {"next run": task.next_run,
+                    "latest run id": latest_run_id,
+                    "start time": task.sctool.get_table_value(parsed_table=task_history, column_name="start time",
+                                                              identifier=latest_run_id),
+                    "end time": task.sctool.get_table_value(parsed_table=task_history, column_name="end time",
+                                                            identifier=latest_run_id),
+                    "duration": task.sctool.get_table_value(parsed_table=task_history, column_name="duration",
+                                                            identifier=latest_run_id)}
+    return task_details
+
+
+def validate_previous_task_details(task, previous_task_details):
+    """
+    Compares the details of the task next run and history to the previously extracted next run and history
+    """
+    current_task_details = wait_until_task_finishes_return_details(task, wait=False)
+    for detail_name in current_task_details:
+        assert current_task_details[detail_name] == previous_task_details[detail_name],\
+            f"previous task {detail_name} is not identical to the current history"
+
+
+def upgrade_scylla_manager(pre_upgrade_manager_version, target_upgrade_version, manager_node, db_cluster):
+    LOGGER.debug("Stopping manager server")
+    if manager_node.is_docker():
+        manager_node.remoter.run('sudo supervisorctl stop scylla-manager')
+    else:
+        manager_node.remoter.run("sudo systemctl stop scylla-manager")
+
+    LOGGER.debug("Stopping manager agents")
+    for node in db_cluster.nodes:
+        node.remoter.run("sudo systemctl stop scylla-manager-agent")
+
+    LOGGER.debug("Upgrading manager server")
+    manager_node.upgrade_mgmt(target_upgrade_version, start_manager_after_upgrade=False)
+
+    LOGGER.debug("Upgrading and starting manager agents")
+    for node in db_cluster.nodes:
+        node.upgrade_manager_agent(target_upgrade_version)
+
+    LOGGER.debug("Starting manager server")
+    if manager_node.is_docker():
+        manager_node.remoter.run('sudo supervisorctl start scylla-manager')
+    else:
+        manager_node.remoter.run("sudo systemctl start scylla-manager")
+    time_to_sleep = 30
+    LOGGER.debug('Sleep {} seconds, waiting for manager service ready to respond'.format(time_to_sleep))
+    sleep(time_to_sleep)
```

Comment:
i thought about it, shouldn't we have a function to check when manager-server is fully up (like checking in logs for a message it started up) instead of waiting?
it because in a slow system, 30 seconds will not be enough, and in a fast system, we are just causing a delay.

Reply:
@ShlomiBalalis , have you thought about this one?

Reply:
We can check if the port is occupied, like we do with scylla. Though, I don't know it it's really necessary

Reply:
just a matter getting rid of this sleep, where we could have a much better mechanism in place

Reply:
I mean... I agree, it'll be better, but I don't want to stall this pr on this one.
I added a ticket for it - https://trello.com/c/dfs94ACI/1979-add-a-waituntilmanagerup-function

Reply:
great, thank you